src/data_import/database_reader.py

```diff
@@ -0,0 +1,57 @@
+# TODO we need some class which help with reading data from database
+
+from database import Reader
+from typing import List, Optional
+from data_import import AVAILABLE_RAM_MEMORY, SIZE_OF_VALUE
+import pandas as pd
+
+
+class DatabaseReader:
+    def __init__(self, db_name: str, coll_name: str):
+        self.error = ''
+        self.need_chunks = False
+        # TODO check size od collection, get number of records
+        size = 10
+        try:
+            self.database = Reader(db_name, coll_name)
+            # TODO get columns name of collection
+            self.columns_name = list(self.database.get_nth_chunk('', []).columns)
+            self.need_chunks = size > self.get_chunksize()
+        except Exception as e:
+            print(e)
+            self.error = 'There is some problem with database. Please try again.'
+        self.reader = None
+
+    def get_columns_name(self) -> List[str]:
+        return self.columns_name
+
+    def get_error(self) -> str:
+        return self.error
+
+    # approximate size of chunk, we want using ram as good as possible
+    def get_chunksize(self) -> int:
+        return AVAILABLE_RAM_MEMORY // (len(self.columns_name) * SIZE_OF_VALUE)
+
+    def is_file_big(self):
+        return self.need_chunks
+
+    def read(self, columns: Optional[List[str]]):
+        if columns is None:
+            columns = self.columns_name
+        if self.need_chunks:
+            self._read_by_chunks(columns)
```

Comment:
I am not sure how this part is supposed to work. __read_by_chunks_ is a Generator, but here it's not being iterated on, nor is it assigned to self.reader or being returned?